extension/README.md.args.mjs

```diff
@@ -42,110 +36,342 @@ yarn start
 
 ---
 
-## Checkpoint 1: 🔏 Staking 💵
+## 🧑‍🚀 Your Mission
+
+🏦 Build a \`CrowdFund.sol\` contract that collects **ETH** from numerous addresses using a payable \`contribute()\` function and keeps track of \`balances\`. After some \`deadline\` if it has at least some \`threshold\` of ETH, it sends it to a \`FundingRecipient\` contract (This is a stand-in for any potential use case a group of people would want to fund together). It then triggers the \`complete()\` action, sending the full balance. If not enough **ETH** is collected, allow users to \`withdraw()\`.
 
-You'll need to track individual \`balances\` using a mapping:
+🔢 Each step is laid out in the following checkpoints. Try to complete them without hints but if you are struggling then you can get clearer context by pressing the  "🔎 Hint" in each checkpoint.
+
+👀 Also, you should try to keep your contract organized by the standard you will see in the contract. Keeping errors, events, functions, etc. sorted under their own sections helps to maintain contract readability .
+## Checkpoint 1: 🤝 Contributing 💵
+
+>Let's start by implementing a state variable that we will need in the function logic.
+
+⚖️ You'll need to track individual \`balances\` using a mapping. This way we will know who gave what in the event that the funding effort fails to raise enough and everyone needs to be refunded. Add it under the existing fundingRecipient variable.
 
 \`\`\`solidity
-mapping ( address => uint256 ) public balances;
+mapping(address => uint256) public balances;
 \`\`\`
 
-And also track a constant \`threshold\` at \`1 ether\`
+> Next let's add an event. Events are useful for outside services that are watching the chain  for certain things to occur. In our case, the front end is going to use this event to know when a contribution takes place.
+
+📣 Add an event to the contract called \`Contribution\` that receives the address of the contributor and the amount they contributed.
 
 \`\`\`solidity
-uint256 public constant threshold = 1 ether;
+event Contribution(address, uint256);
+\`\`\`
+
+ >📝 Note: If you use named arguments in your event (e.g. \`event Contribution(address indexed contributor, uint256 amount)\`), you'll need to update \`/packages/nextjs/app/contributions/page.tsx\` to reference event parameters by their names instead of numeric indices.
+
+### Implementing the \`contribute()\` function
+
+> 👩‍💻 Now focus on writing your \`contribute()\` function. The payable method already exists but is empty. Go fill it with logic!
+
+The goal of this function is to allow anyone to contribute to the pool of funds. To do this effectively it will need to do the following:
+- Update the \`balances\` mapping
+- Emit the \`Contribute\` event 
+
+<details markdown='1'>
+<summary>🔎 Hint</summary>
+
+You can set mappings like you would access a Javascript array.
+For a mapping like this \`mapping(address => uint256) public dir\` you would access is like this:
```

Comment:
Maybe it's better to use something like to avoid `+= 1` below

For a mapping like this `mapping(address => uint256) public dir` and `address addr = 0x1234...5678`  you would access it like this: `dir[addr]`